RetroBar/Themes/System.xaml.cs

```diff
@@ -0,0 +1,37 @@
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Media;
+
+namespace RetroBar
+{
+    public partial class SystemTheme : ResourceDictionary
+    {
+        private static void InvalidateMeasureRecursively(UIElement main)
```

Comment:
Rather than this hack, a cleaner approach would be similar to how the taskbar handles changes to AllowTransparency.

Reply:
Here's the approach I tested as working:
- Rather than binding FlowDirection on the taskbar window, set it in code in the constructor
- In the Settings_PropertyChanged method, add a case for Language and check if flow_direction has changed.
- If flow_direction has changed, then either perform invalidation like this, or simply recreate the taskbar window like we do with AllowsTransparency changes.

That should allow us to keep code out of the theme.

Reply:
Alright, but are you sure it will help me to fix the issue of not updating the context menu size, I will try to see that alternative.

Reply:
Yeah, I was able to have a correct context menu size with this approach. I was switching between English RTL and Russian LTR for testing.

Reply:
Okay, I may take a bit, because I'm going to be out for a while, I'll see this when I get back.

Reply:
It works, but it closes and opens all the taskbars again, as if reloading, I think this can be a bit annoying, that from the code behind System theme could also run the extraction of system resources like aero.msstyles icons and images, or similar, but anyway I will make this change.

Reply:
As you suggested, I can change it to invalidation instead of recreating the taskbar, I think it can be done in a future commit, but now I don't want to touch this code, at least not while it works. But if you think it's better you can make this change if you are also uncomfortable with recreating the taskbar...

Reply:
@dremin, this issue is happening again when switching between Hebrew and Arabic, it is easily noticeable in the tooltips, maybe the old invalidation method is necessary because we should not reload the taskbar when switching to any language in the settings.